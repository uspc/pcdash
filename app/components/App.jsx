import React from 'react';
import uuid from 'uuid';
import connect from '../libs/connect';
import Lanes from './Lanes';
import LaneActions from '../actions/LaneActions';
import {compose} from 'redux';
import {DragDropContext} from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import PageHeader from './PageHeader';
import Bootstrap from 'bootstrap/dist/css/bootstrap.css';
import { ContextMenu, MenuItem, ContextMenuLayer } from 'react-contextmenu';


// changed export const app to const app.. since we are not importing at app level i.e import {App} from App.js
const App = ({LaneActions, lanes}) => {
  const addLane = () => {
    LaneActions.create({
      id: uuid.v4(),
      name: 'New lane'
    });
  };

  return (

      <div>
          <nav className="navbar navbar-inverse navbar-fixed-top">
              <div className="container-fluid">
                  <div className="navbar-header">
                      <a className="navbar-brand" href="#">BuildIt</a>
                  </div>
                  <ul className="nav navbar-nav">
                      <li className="active"><a id="a1" href="#" onClick={addLane}>Add Lane</a></li>
                  </ul>
                  <ul className="nav navbar-nav navbar-right">
                      <li><a href="#"><span className="glyphicon glyphicon-user"></span> Sign Up</a></li>
                      <li><a href="#"><span className="glyphicon glyphicon-log-in"></span> Login</a></li>
                  </ul>
              </div>
          </nav>

          <div className="test"/>

          <Lanes lanes={lanes} />

        {/*
         <div className="well-lg" onClick={addLane} >Add Lanes</div>
        <div id="my-context-menu">
            <MyComponent  />
            <MyContextMenu />
        </div>
         */}

        {/* <button className='add-lane' onClick={addLane}>+</button> */}


        <div className="navbar-fixed-bottom">Â© 2016 Buildit. All rights reserved.</div>
    </div>

  );
};


export default compose(
    DragDropContext(HTML5Backend),
    connect(({lanes}) => ({
      lanes
    }), {
      LaneActions
    })
)(App)





