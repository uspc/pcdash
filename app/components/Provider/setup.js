import storage from '../../libs/storage';
import persist from '../../libs/persist';
import NoteStore from '../../stores/NoteStore';
import LaneStore from '../../stores/LaneStore';
import loadseed from '../../libs/loadseed';

export default alt => {
  alt.addStore('NoteStore', NoteStore);
  alt.addStore('LaneStore', LaneStore);

  loadseed(sessionStorage,'app',alt);
  //persist(alt, storage(sessionStorage), 'app');
}


