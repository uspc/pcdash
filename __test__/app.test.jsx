//import renderer from 'react-test-renderer';
//import {App} from '../app/components/App.jsx';
//import Lane from '../app/components/Lane';
//import React from 'react';
//
//test('Sample jest invoke testing', () => {
//    expect(1).toBe(1);
//});
//
//test('Link renders correctly', () => {
//    const tree = renderer.create(<App lanes={[]}/>).toJSON();
//    expect(tree).toMatchSnapshot();
//});



import renderer from 'react-test-renderer';
import App from '../app/components/App.jsx';
import Lane from '../app/components/Lane';
import React from 'react';
import Provider from '../app/components/Provider'

test('Sample jest invoke testing', () => {
    expect(1).toBe(1);
});

test('Link renders correctly', () => {
    const tree = renderer.create(<Provider><App /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});


