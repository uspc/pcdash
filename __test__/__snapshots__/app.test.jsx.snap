exports[`test Link renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-inverse navbar-fixed-top">
    <div
      className="container-fluid">
      <div
        className="navbar-header">
        <a
          className="navbar-brand"
          href="#">
          BuildIt
        </a>
      </div>
      <ul
        className="nav navbar-nav">
        <li
          className="active">
          <a
            href="#"
            id="a1"
            onClick={[Function]}>
            Add Lane
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right">
        <li>
          <a
            href="#">
            <span
              className="glyphicon glyphicon-user" />
             Sign Up
          </a>
        </li>
        <li>
          <a
            href="#">
            <span
              className="glyphicon glyphicon-log-in" />
             Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="lanes" />
  <div
    className="navbar-fixed-bottom">
    Â© 2016 Buildit. All rights reserved.
  </div>
</div>
`;
